src/lazy/binary/raw/v1_1/struct.rs

```diff
@@ -237,6 +237,11 @@ impl<'top> RawBinaryStructIterator_1_1<'top> {
         BinaryBuffer<'top>,
     )> {
         let opcode = buffer.expect_opcode()?;
+
+        if opcode.is_delimited_end() {
+            return Ok((None, buffer));
+        }
+
```

Comment:
nit: this could be worked into the if-else below. It's not really any more special than the `NOP` handling.

Should this return:
```rust
Ok((None, buffer.consume(1)))
```
If the original ByteBuffer is setup correctly, then the end of container marker should be the last byte in the buffer. So moving beyond it, should leave us with an empty buffer, which will be a quick check if someone tries to read again.